kroxylicious/src/main/java/io/kroxylicious/proxy/internal/net/EndpointResolver.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.net;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Used by the {@link io.kroxylicious.proxy.internal.KafkaProxyInitializer} to resolve incoming channel
+ * metadata into a {@link VirtualClusterBinding}.
+ */
+public interface EndpointResolver {
```

Comment:
Is this an Endpoint resolver (resolves something using endpoints)? Or a `VirtualClusterBindingResolver` that takes some input and resolves a VirtualClusterBinding? Not sure which is easier to read or more standard. Maybe VirtualClusterBindingResolver is better since it's not just Endpoint that's determining the VCB but Endpoint plus routing data? We also have routing term around a little bit, could this be a VirtualClusterRouter?

Reply:
Agreed, `VirtualClusterBindingResolver` would be superior to `EndpointResolver`. `VirtualClusterRouter` would also work.